internal/characters/dehya/burst.go

```diff
@@ -0,0 +1,194 @@
+
+func init() {
+	//TODO:Deprecate bursty frames in favor of a constant?
+	burstFrames = frames.InitAbilSlice(102) // Q -> E/D/J
+	burstFrames[action.ActionSwap] = 102    // Q -> Swap
+
+	kickFrames = frames.InitAbilSlice(72)       // Q -> Dash/Walk
+	kickFrames[action.ActionAttack] = 75        // Q -> N1
+	kickFrames[action.ActionSkill] = 71         // Q -> E
+	kickFrames[action.ActionJump] = 73          // Q -> J
+	kickFrames[action.ActionSwap] = kickHitmark //Q -> Swap
+
+}
+
+func (c *char) Burst(p map[string]int) action.ActionInfo {
+	burstIsJumpCancelled = false
+	ai := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Flame-Mane's Fist",
+		AttackTag:  attacks.AttackTagElementalBurst,
+		ICDTag:     attacks.ICDTagElementalBurst,
+		ICDGroup:   attacks.ICDGroupDefault,
+		StrikeType: attacks.StrikeTypeDefault,
+		Element:    attributes.Pyro,
+		Durability: 25,
+		Mult:       burstPunchAtk[c.TalentLvlBurst()],
+		FlatDmg:    (c.c1var[0] + burstPunchHP[c.TalentLvlBurst()]) * c.MaxHP(),
+	}
+
+	c.c6count = 0
+	remainingFieldDur = 0
+	if c.StatusIsActive(dehyaFieldKey) {
+		// pick up field at start
+		remainingFieldDur = c.StatusExpiry(dehyaFieldKey) + sanctumPickupExtension - c.Core.F //dur gets extended on field recast by a low margin, apparently
+		c.Core.Log.NewEvent("sanctum removed", glog.LogCharacterEvent, c.Index).
+			Write("Duration Remaining ", remainingFieldDur+sanctumPickupExtension).
+			Write("DoT tick CD", c.StatusDuration(skillICDKey))
+		c.DeleteStatus(dehyaFieldKey)
+	}
+
+	c.Core.Tasks.Add(func() {
+		c.AddStatus(burstKey, 240, false)
+		c.burstCast = c.Core.F
+		c.punchSrc = true
+		c.burstCounter = 0
+		c.burstPunch(c.punchSrc, true)
+	}, burstDoT1Hitmark)
+
+	c.Core.QueueAttack(
+		ai,
+		combat.NewBoxHitOnTarget(c.Core.Combat.Player(), geometry.Point{Y: -2.8}, 5, 7.8),
+		burstDoT1Hitmark,
+		burstDoT1Hitmark,
+		c.c4cb(),
+		c.c6cb(),
+	)
+
+	c.ConsumeEnergy(15) //TODO: If this is ping related, this could be closer to 1 at 0 ping
+	c.SetCDWithDelay(action.ActionBurst, 18*60, 1)
+
+	return action.ActionInfo{
+		Frames:          frames.NewAbilFunc(burstFrames),
+		AnimationLength: burstFrames[action.ActionAttack],
+		CanQueueAfter:   burstFrames[action.ActionAttack], // earliest cancel
+		State:           action.BurstState,
+	}
+}
+
+func (c *char) burstPunch(src bool, auto bool) action.ActionInfo {
+	hitmark := punchSlowHitmark
+	if !auto {
+		hitmark = punchHitmarks[c.burstCounter]
+	}
+
+	ai := combat.AttackInfo{
+		ActorIndex: c.Index,
+		Abil:       "Flame-Mane's Fist",
+		AttackTag:  attacks.AttackTagElementalBurst,
+		ICDTag:     attacks.ICDTagElementalBurst,
+		ICDGroup:   attacks.ICDGroupDefault,
+		StrikeType: attacks.StrikeTypeDefault,
+		Element:    attributes.Pyro,
+		Durability: 25,
+		Mult:       burstPunchAtk[c.TalentLvlBurst()],
+		FlatDmg:    (c.c1var[0] + burstPunchHP[c.TalentLvlBurst()]) * c.MaxHP(),
+	}
+
+	c.Core.Tasks.Add(func() {
+		if c.punchSrc != src {
+			return
+		}
```

Comment:
Should this instead be a counter or what frame the call was from or something? If I am reading the code correctly auto case will have `punchSrc` true whereas skill/attack will have `punchSrc` false. Current solution not clear and not clear if bug free. IE: if my skill/attack early trigger happens before the "slow" attack, this task will set `punchSrc` back to `true` making the first slow attack now valid and generating an attack when it shouldnt.